hibernate-core/src/main/java/org/hibernate/sql/results/graph/embeddable/AbstractEmbeddableInitializer.java

```diff
@@ -409,15 +425,22 @@ private Object determineParentInstance(RowProcessingState processingState) {
 			return null;
 		}
 
-		final Initializer parentInitializer = processingState.resolveInitializer( parentPath );
+		return processingState.resolveInitializer( parentPath );
+	}
+
+	private Object determineParentInstance(Initializer parentInitializer) {
 
 		if ( parentInitializer.isCollectionInitializer() ) {
 			return ( (CollectionInitializer) parentInitializer ).getCollectionInstance().getOwner();
 		}
 
 		final EntityInitializer parentEntityInitializer = parentInitializer.asEntityInitializer();
 		if ( parentEntityInitializer != null ) {
-			return parentEntityInitializer.getEntityInstance();
+			return parentEntityInitializer.getInitializedInstance();
+		}
+
+		if ( navigablePath.getParent() == null ) {
```

Comment:
Would be great if you could add a code comment that explains what the purpose of this if block is.

Reply:
not sure how to explain, tried not to change the previous behaviour of the `determineParentInstance` that in case of 
```
// Otherwise, fallback to determining the parent-initializer by path
		//		todo (6.0) - this is the part that should be "subsumed" based on the
		//			comment above

		final NavigablePath parentPath = navigablePath.getParent();
		if ( parentPath == null ) {
			return null;
		}
```
returned null for the parent instance value.

Reply:
Hmm, ok I understand now. Maybe you can do an experiment and comment out this if block to see if we still need this code. It seems strange to me that an embeddable initializer could have a null parent navigable path.

Reply:
by the way the code is wrong, if `parentInitializer` is null the call made few line before would cause a NPE

Reply:
not sure too, 2 oprions
```
private Object determineParentInstance(Initializer parentInitializer) {
		if ( parentInitializer == null ) {
			if ( navigablePath.getParent() == null ) {
				/*
				 In case we were not able to find the parentInitializer and the navigablePath.getParent() is null then the parent instance is  considered null;
				 Not sure this case can happen
				 */
				return null;
			}
			else {
				throw new UnsupportedOperationException("Cannot determine Embeddable: " + navigablePath  + " parent instance" );
			}
		}

		if ( parentInitializer.isCollectionInitializer() ) {
			return ( (CollectionInitializer) parentInitializer ).getCollectionInstance().getOwner();
		}

		final EntityInitializer parentEntityInitializer = parentInitializer.asEntityInitializer();
		if ( parentEntityInitializer != null ) {
			return parentEntityInitializer.getInitializedInstance();
		}

		throw new UnsupportedOperationException( "The Embeddable: " + navigablePath + " parent initializer is neither an instance of an EntityInitializer nor of a CollectionInitializer" );
	}
```

or 
```
private Object determineParentInstance(Initializer parentInitializer) {
		if ( parentInitializer == null ) {
			
				throw new UnsupportedOperationException("Cannot determine Embeddable: " + navigablePath  + " parent instance" );
			
		}

		if ( parentInitializer.isCollectionInitializer() ) {
			return ( (CollectionInitializer) parentInitializer ).getCollectionInstance().getOwner();
		}

		final EntityInitializer parentEntityInitializer = parentInitializer.asEntityInitializer();
		if ( parentEntityInitializer != null ) {
			return parentEntityInitializer.getInitializedInstance();
		}

		throw new UnsupportedOperationException( "The Embeddable: " + navigablePath + " parent initializer is neither an instance of an EntityInitializer nor of a CollectionInitializer" );
	}
```